recipes/http2-priority-signaling/src/workflow.ts

```diff
@@ -0,0 +1,363 @@
+
+			let hasPriority = false;
+			let allPriority = true;
+
+			for (const pair of pairs) {
+				const keyNode = pair.find({
+					rule: {
+						kind: "property_identifier",
+						regex: "^priority$",
+					},
+				});
+
+				if (keyNode) {
+					hasPriority = true;
+				} else {
+					allPriority = false;
+				}
+			}
+
+			if (allPriority && hasPriority) {
+				// Remove the entire object argument from the call
+				const callText = call.text();
+				const objText = obj.text();
+				// Use s flag to match across newlines (including the multiline object)
+				const cleanedCall = callText.replace(new RegExp(`(,\\s*)?${escapeRegex(objText)}(,\\s*)?`, "s"), "");
+				const finalCall = cleanedCall.replace(/,\s*\)/, ")");
+
+				if (finalCall !== callText) {
+					edits.push(call.replace(finalCall));
+				}
+			} else if (hasPriority) {
+				// Object has other properties, so just remove priority pair
+				edits.push(...removePriorityPairFromObject(obj));
+			}
+		}
+	}
+
+	return edits;
+}
+
+/**
+ * Remove priority property from session.request() call options
+ */
+function removeRequestPriority(rootNode: SgNode<Js>): Edit[] {
+	const edits: Edit[] = [];
+
+	// Find all request calls and clean priority from their options
+	const requestCalls = rootNode.findAll({
+		rule: {
+			pattern: "$SESSION.request($$$_ARGS)",
+		},
+	});
+
+	for (const call of requestCalls) {
+		const objects = call.findAll({
+			rule: {
+				kind: "object",
+			},
+		});
+
+		for (const obj of objects) {
+			// Check if object only contains priority properties
+			// Get immediate children pairs only (not nested)
+			const pairs = obj.children().filter((child) => child.kind() === "pair");
+
+			let hasPriority = false;
+			let allPriority = true;
+
+			for (const pair of pairs) {
+				const keyNode = pair.find({
+					rule: {
+						kind: "property_identifier",
+						regex: "^priority$",
+					},
+				});
+
+				if (keyNode) {
+					hasPriority = true;
+				} else {
+					allPriority = false;
+				}
+			}
+
+			if (allPriority && hasPriority) {
+				// Remove the entire object argument from the call
+				const callText = call.text();
+				const objText = obj.text();
+				const cleanedCall = callText.replace(new RegExp(`(,\\s*)?${escapeRegex(objText)}(,\\s*)?`, "s"), "");
+				const finalCall = cleanedCall.replace(/,\s*\)/, ")");
+
+				if (finalCall !== callText) {
+					edits.push(call.replace(finalCall));
+				}
+			} else if (hasPriority) {
+				// Object has other properties, so just remove priority pair
+				edits.push(...removePriorityPairFromObject(obj));
+			}
+		}
+	}
+
+	return edits;
+}
+
+/**
+ * Remove entire stream.priority() method calls
+ */
+function removePriorityMethodCalls(
+	rootNode: SgNode<Js>,
+): { edits: Edit[]; linesToRemove: Range[] } {
+	const edits: Edit[] = [];
+	const linesToRemove: Range[] = [];
+
+	const priorityCalls = rootNode.findAll({
+		rule: {
+			pattern: "$STREAM.priority($$$ARGS)",
+		},
+	});
```

Comment:
This will match any object that has a method called priority. I think it’s not safe to run this codemod this way, we need some guardrails to ensure that the session object is really what we expect it to be.